app/services/accounts/accounts_filter_service.rb

```diff
@@ -0,0 +1,32 @@
+module Accounts
+  class AccountsFilterService < ApplicationService
+    def initialize(accounts, params)
+      @accounts = accounts
+      @params = params
+    end
+
+    def call
+      search
+      filter_by_status
+      filter_by_role
```

Comment:
Check params is present to avoid running into these methods and running redundant queries, in case user only filter but not search, without the condition check params search present, it will still run into the search method to run search query.